src/hal/src/pso/descriptor.rs

```diff
@@ -74,7 +74,8 @@ pub struct DescriptorSetLayoutBinding {
     pub count: DescriptorArrayIndex,
     /// Valid shader stages.
     pub stage_flags: ShaderStageFlags,
-    // TODO: immutable samplers?
+    /// Use the associated list of immutable samplers.
+    pub immutable_samplers: bool,
```

Comment:
where is this needed?

Reply:
Resolved in internal discussion.
Vulkan takes a null ptr or array of samplers. Works basically like a Option, but combining the samplers into one iterator.